requests/utils.py

```diff
@@ -38,6 +39,92 @@
 DEFAULT_CA_BUNDLE_PATH = certs.where()
 
 
+class _ListNode(object):
+    def __init__(self, key, val, expired_at, prev=None, nxt=None):
+        self.key = key
+        self.val = val
+        self.expired_at = expired_at
+        self.prev = prev
+        self.nxt = nxt
+
+    def is_expired(self):
+        return time.time() > self.expired_at
+
+    def link_to(self, former, latter):
+        # Return: former <-> self <-> latter
+        self.prev = former
+        self.nxt = latter
+        if former is not None:
+            former.nxt = self
+        if latter is not None:
+            latter.prev = self
+
+    def unlink_self(self):
+        # Original: prev.prev <-> prev <-> self <-> nxt
+        # Returned: prev.prev <-> prev <-> nxt
+        self.prev.link_to(self.prev.prev, self.nxt)
+
+    @staticmethod
+    def dummy_node():
+        return _ListNode(key=None, val=None, expired_at=None)
+
+
+class TTLCache(object):
+    def __init__(self, ttl=3600, maxitems=64):
+        self.ttl = ttl
+        self.key_to_node = {}
+        self.maxitems = maxitems
+
+        self.list_head = _ListNode.dummy_node()
```

Comment:
I'm not sure if the complexity of this linked list is worth its benefits. I'd rather prefer a linear search for the oldest entry when `maxitems` is reached. But please wait for other opinions before you change anything.